awx/main/utils/proxy.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2024 Ansible, Inc.
+# All Rights Reserved.
+
+
+# DRF
+from rest_framework.request import Request
+
+
+"""
+Note that these methods operate on request.environ. This data is from uwsgi.
+It is the source data from which request.headers (read-only) is constructed.
+"""
+
+
+def is_proxy_in_headers(request: Request, proxy_list: list[str], headers: list[str]) -> bool:
```

Comment:
I'd like a little more documentation on what `proxy_list` is here, because its name doesn't match well with what it actually seems to be. It seems that it corresponds to the proxy allow list setting? So this is a list of IPs we will accept as proxies?

Reply:
And why call it `headers` here and `remote_headers` in other places?

So this is like a list of the keys of headers, that may specify the IP that it got forwarded from?

Reply:
> I'd like a little more documentation on what proxy_list is here, because its name doesn't match well with what it actually seems to be. It seems that it corresponds to the proxy allow list setting? So this is a list of IPs we will accept as proxies?
A list of ips or hostnames. It's more than _likely_ ips. Some customer _might_ put a hostname to we would match against the special case `REMOTE_HOST` (described below).

```
Any security-related use of X-Forwarded-For (such as for rate limiting or IP-based access control) must only use IP addresses added by a trusted proxy. Using untrustworthy values can result in rate-limiter avoidance, access-control bypass, memory exhaustion, or other negative security or availability consequences.
```

> And why call it headers here and remote_headers in other places?
No reason. To add the confusion even more there are 2 special entries that actually aren't headers. `REMOTE_ADDR` and `REMOTE_HOST`. They are more like env vars, set by uwsgi, but they are in this `request.environ` dict. They don't start with `HTTP_`, so that distinguishes them in Django land.

> So this is like a list of the keys of headers, that may specify the IP that it got forwarded from?
yep